[Bug]: CDC replicate stream can crash proxy when replaying transaction system messages

## Environment

- **Milvus Version**: `master-20260524-34937a5` (`34937a5f7892ae14df9430a7cef2cf5d2809619e`)
- **Deployment Mode**: cluster, CDC two-cluster E2E
- **MQ**: pulsar
- **SDK**: CDC Python E2E / pymilvus
- **OS**: Kubernetes on QA Jenkins environment

## Reproduction

### Option B: Steps (for systemic / stability / timing issues)

1. Deploy two Milvus clusters with streaming enabled and CDC replication configured.
2. Run the CDC network fault E2E job on master:
   - https://qa-jenkins.milvus.io/job/milvus_cdc_network_fault_test/5/
   - image: `harbor.milvus.io/manta/milvus:master-20260524-34937a5`
3. Let CDC replicate DDL and DML traffic from upstream to downstream.
4. The upstream CDC sender sends a sequence containing a broadcast DDL followed by transaction messages on the same pchannel.
5. The downstream proxy enters CrashLoopBackOff.

### Trigger Conditions

- **Frequency**: reproducible in latest CDC master E2E runs that hit transaction replay through CDC replicate stream.
- **First observed after**: master CDC scheduled jobs; the problematic code path appears to have been introduced by PR #47179 (`446f06eb021`).
- **Does NOT happen when**: the previous successful comparison jobs were on the `2.6` branch, so they are not a valid master baseline.

## Expected Behavior

CDC replicate stream should be able to replay already committed WAL messages, including replicated transaction system messages such as `BeginTxn` and `CommitTxn`, without crashing proxy. These messages should reach the streamingnode WAL replicate handling path.

## Actual Behavior

The downstream proxy panics when `Replicate().Append()` internally calls the ordinary client append path. `AppendMessages()` calls `assertValidMessage()`, which rejects system messages. Replicated `BeginTxn` / `CommitTxn` are valid for CDC replay, but they are rejected before reaching the WAL replicate interceptor.

## Error Logs

From `milvus_cdc_network_fault_test #5`, downstream proxy:

```text
panic: system message is not allowed to append from client

github.com/milvus-io/milvus/internal/distributed/streaming.assertValidMessage
    /workspace/milvus/internal/distributed/streaming/append.go:30
github.com/milvus-io/milvus/internal/distributed/streaming.(*walAccesserImpl).AppendMessages
    /workspace/milvus/internal/distributed/streaming/util.go:23
github.com/milvus-io/milvus/internal/distributed/streaming.replicateService.Append
    /workspace/milvus/internal/distributed/streaming/replicate_service.go:56
github.com/milvus-io/milvus/internal/proxy/replicate.(*ReplicateStreamServer).handleReplicateMessage
    /workspace/milvus/internal/proxy/replicate/replicate_stream_server.go:129
```

A CDC sender log around the crash shows the replayed sequence:

```text
type=CreateIndex messageID=4/29/0 lastConfirmed=4/27/0
type=BeginTxn    messageID=4/32/0 lastConfirmed=4/29/0 txnID=466526581250004390
type=Insert      messageID=4/34/0 lastConfirmed=4/29/0 txnID=466526581250004390
type=Insert      messageID=4/35/0 lastConfirmed=4/29/0 txnID=466526581250004390
type=Insert      messageID=4/36/0 lastConfirmed=4/29/0 txnID=466526581250004390
type=CommitTxn   messageID=4/37/0 lastConfirmed=4/29/0 txnID=466526581250004390
```

## Non-default Configuration

CDC replication between two Milvus clusters in the QA CDC E2E pipeline.

## Analysis Hints (Optional)

Suspected regression:

- PR: #47179
- Commit: `446f06eb02199c05b2e7324b3103e41a2a7a57de`
- Change: `replicateService.Append()` was changed from a direct WAL append path to `AppendMessages(ctx, msg)`.

Current code path:

```go
// internal/distributed/streaming/replicate_service.go
resp := s.AppendMessages(ctx, msg)
```

`AppendMessages()` is the ordinary client append helper and starts with `assertValidMessage(msgs...)`, which intentionally rejects system messages. That protection should remain for normal client appends, but CDC replicate replay needs a separate internal append path that can write replicated `BeginTxn` / `CommitTxn` messages.

--- a/internal/distributed/streaming/replicate_service.go
+++ b/internal/distributed/streaming/replicate_service.go
@@ -53,8 +53,7 @@ func (s replicateService) Append(ctx context.Context, rmsg message.ReplicateMuta
 	if err != nil {
 		return nil, err
 	}
-	resp := s.AppendMessages(ctx, msg)
-	return resp.Responses[0].AppendResult, resp.Responses[0].Error
+	return s.appendReplicateMessageToWAL(ctx, msg)
 }
 
 func (s replicateService) UpdateReplicateConfiguration(ctx context.Context, req *milvuspb.UpdateReplicateConfigurationRequest) error {

--- a/internal/distributed/streaming/replicate_service_test.go
+++ b/internal/distributed/streaming/replicate_service_test.go
@@ -90,6 +90,68 @@ func TestReplicateService(t *testing.T) {
 	}
 }
 
+func TestReplicateServiceAppendTxnSystemMessage(t *testing.T) {
+	c := mock_client.NewMockClient(t)
+	as := mock_client.NewMockAssignmentService(t)
+	c.EXPECT().Assignment().Return(as).Maybe()
+
+	h := mock_handler.NewMockHandlerClient(t)
+	p := mock_producer.NewMockProducer(t)
+	var appended bool
+	p.EXPECT().Append(mock.Anything, mock.Anything).RunAndReturn(func(ctx context.Context, mm message.MutableMessage) (*types.AppendResult, error) {
+		appended = true
+		assert.Equal(t, message.MessageTypeBeginTxn, mm.MessageType())
+		assert.Equal(t, "by-dev-rootcoord-dml_0_1v0", mm.VChannel())
+		return &types.AppendResult{
+			MessageID: walimplstest.NewTestMessageID(1),
+			TimeTick:  1,
+		}, nil
+	}).Maybe()
+	p.EXPECT().IsAvailable().Return(true).Maybe()
+	p.EXPECT().Available().Return(make(chan struct{})).Maybe()
+	h.EXPECT().CreateProducer(mock.Anything, mock.Anything).Return(p, nil).Maybe()
+
+	as.EXPECT().GetReplicateConfiguration(mock.Anything).Return(replicateutil.MustNewConfigHelper(
+		"by-dev",
+		&commonpb.ReplicateConfiguration{
+			Clusters: []*commonpb.MilvusCluster{
+				{ClusterId: "primary", Pchannels: []string{"primary-rootcoord-dml_0"}},
+				{ClusterId: "by-dev", Pchannels: []string{"by-dev-rootcoord-dml_0"}},
+			},
+			CrossClusterTopology: []*commonpb.CrossClusterTopology{
+				{SourceClusterId: "primary", TargetClusterId: "by-dev"},
+			},
+		},
+	), nil)
+
+	rs := &replicateService{
+		walAccesserImpl: &walAccesserImpl{
+			lifetime:             typeutil.NewLifetime(),
+			clusterID:            "by-dev",
+			streamingCoordClient: c,
+			handlerClient:        h,
+			producers:            make(map[string]*producer.ResumableProducer),
+		},
+	}
+
+	beginMsg := message.NewBeginTxnMessageBuilderV2().
+		WithHeader(&message.BeginTxnMessageHeader{}).
+		WithBody(&message.BeginTxnMessageBody{}).
+		WithVChannel("primary-rootcoord-dml_0_1v0").
+		MustBuildMutable()
+	immutableMsg := beginMsg.WithLastConfirmedUseMessageID().WithTimeTick(1).IntoImmutableMessage(pulsar2.NewPulsarID(
+		pulsar.NewMessageID(1, 2, 3, 4),
+	))
+	replicateMsg := message.MustNewReplicateMessage("primary", immutableMsg.IntoImmutableMessageProto())
+
+	var err error
+	assert.NotPanics(t, func() {
+		_, err = rs.Append(context.Background(), replicateMsg)
+	})
+	assert.NoError(t, err)
+	assert.True(t, appended)
+}
+
 func TestReplicateService_GetReplicateConfiguration(t *testing.T) {
 	t.Run("success", func(t *testing.T) {
 		c := mock_client.NewMockClient(t)

--- a/internal/distributed/streaming/util.go
+++ b/internal/distributed/streaming/util.go
@@ -70,6 +70,15 @@ func (w *walAccesserImpl) AppendMessages(ctx context.Context, msgs ...message.Mu
 	return resp
 }
 
+func (w *walAccesserImpl) appendReplicateMessageToWAL(ctx context.Context, msg message.MutableMessage) (*types.AppendResult, error) {
+	guard, err := w.getProducer(msg.VChannel()).BeginProduce(ctx, msg)
+	if err != nil {
+		return nil, err
+	}
+	resp := producer.BatchCommitProduce(ctx, guard)
+	return resp.Responses[0].AppendResult, resp.Responses[0].Error
+}
+
 // dispatchMessages dispatches the messages into different vchannel.
 func (w *walAccesserImpl) dispatchMessages(msgs ...message.MutableMessage) (map[string][]message.MutableMessage, map[string][]int) {
 	dispatchedMessages := make(map[string][]message.MutableMessage, 0)

